openshift-maven-plugin/plugin/src/test/java/org/eclipse/jkube/maven/plugin/mojo/build/OpenShiftResourceMojoTest.java

```diff
@@ -33,75 +31,57 @@
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
 
-import java.io.File;
-import java.io.IOException;
 import java.util.Collections;
 import java.util.Properties;
 
 import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
 public class OpenShiftResourceMojoTest {
   @Rule
   public TemporaryFolder temporaryFolder = new TemporaryFolder();
 
-  @Mocked
-  private MavenProject mockedMavenProject;
-
-  @Mocked
-  private MavenSession mockedMavenSession;
-
-  @Mocked
-  private Settings mockedSettings;
-
-  @Mocked
-  private JKubeServiceHub mockedJKubeServiceHub;
-
-  @Mocked
   private ClusterAccess mockedClusterAccess;
-
-  @Mocked
-  private KitLogger mockedLogger;
-
-  @Mocked
-  private MavenProjectHelper mockedMavenProjectHelper;
-
-  @Mocked
   private ImageConfigResolver mockedImageConfigResolver;
 
   private OpenshiftResourceMojo openShiftResourceMojo;
 
   @Before
-  public void setUp() throws IOException {
+  public void setUp() {
+    mockedClusterAccess = mock(ClusterAccess.class, RETURNS_DEEP_STUBS);
```

Comment:
I feel like there is too many nested mocks here which may not be needed if you use `RETURNS_DEEP_STUBS` and or `@InjectMocks`

Reply:
Thanks, you're right. I was able to remove unnecessary mocks for JKubeConfiguration class by using `RETURNS_DEEP_STUBS`